### PR TITLE
fix: fixed empty case and overwrite behavior in goldentest new-test

### PR DESCRIPTION
**Pull request by capri-xiyue**
1. fixed empty case when there is no inputs or built in vars
2. fixed overwrite behavior, previously it can't handle truncate case.

### CODE REVIEW

File: templates/model/goldentest/v1beta3/goldentest.go [hunk 1]
```diff
@@ -54,6 +54,14 @@ type Test struct {
 	BuiltinVars []*VarValue `yaml:"builtin_vars,omitempty"`
 }
 
+// TestCase represents a test.yaml describing test configs.
```
Comment (drevell) [inline templates/model/goldentest/v1beta3/goldentest.go:57]: It's not immediately obvious to me why we need both Test and TestCase

**Review by drevell — CHANGES_REQUESTED** [review]

File: templates/model/goldentest/v1beta3/goldentest.go (hunk 1 above)
Comment (capri-xiyue) [reply templates/model/goldentest/v1beta3/goldentest.go:57]: If we don't have `TestCase` here and use the `Test`, as the the inputs and built in vars of `Test` can be empty, yaml marshal will emit a string as `{}`. Therefore I use a `TestCase` here with the api version and kind that won't be empty. Another approach would be use the `Test` but I need to check whether it is a empty struct or not and then use yaml marshal, I feel it is kind of less clean code.

Comment (drevell) [reply templates/model/goldentest/v1beta3/goldentest.go:57]: I sent you https://github.com/abcxyz/abc/pull/390 as an alternative approach to addressing this, let me know what you think
